packages/jsii-pacmak/lib/targets/java.ts

```diff
@@ -492,6 +493,22 @@ class JavaGenerator extends Generator {
         );
         this.code.closeFile('pom.xml');
 
+        /**
+         * Combines a version number with an optional suffix. If the suffix starts with '-' or '.', it will be
+         * concatenated as-is to the semantic version number. Otherwise, it'll be appended to the version number with an
+         * intercalar '-'.
+         *
+         * @param version the semantic version number
+         * @param suffix  the suffix, if any.
+         */
+        function makeVersion(version: string, suffix?: string): string {
+            if (!suffix) { return version; }
```

Comment:
Seems over-designed, I would simply fail if the suffix does not begin with "-" or "."

Reply:
I don't like this, would rather *always* prepend a `-`, but it seems controversial enough that I want to be open to either, while making it easier to go with what I would prefer :D